game/src/core/components/DrawComponent.java

```diff
@@ -156,23 +156,27 @@ public void currentAnimation(final IPath... animationName) {
 
     /**
      * Queue up an Animation to be considered as the next played Animation. Animations are given as
-     * an IPath Array, or multiple variables. Animation length is set as one frame. If you need to
-     * queue longer Animations use {@link #queueAnimation(int, IPath...)}
+     * an IPath Array or multiple variables. Animation length is set to one frame. If you need to
+     * queue longer Animations, use {@link #queueAnimation(int, IPath...)}.
      *
-     * @param next Array of IPaths to Animation
+     * @param next Array of IPaths representing the Animation.
      */
-    public void queueAnimation(IPath... next) {
+    public void queueAnimation(final IPath... next) {
         queueAnimation(1, next);
     }
-
     /**
-     * Queue up an Animation to be considered as the next played Animation. Animations are given as
-     * an IPath Array, or multiple variables. Animation length is set to given parameter.
+     * Queue up an Animation to be considered as the next played Animation.
+     *
+     * <p>Animations are given as an IPath Array or multiple variables. The Animation with the
+     * lowest priority will be added to the queue. If the Animation is already added, nothing will
+     * happen.
      *
-     * @param forFrames number of Frames to play Animation for
-     * @param next Array of IPaths to Animation
+     * <p>Animation length is set to the given parameter.
+     *
+     * @param forFrames Number of frames to play the Animation for.
+     * @param next Array of IPaths representing the Animation.
      */
-    public void queueAnimation(int forFrames, IPath... next) {
+    public void queueAnimation(int forFrames, final IPath... next) {
```

Comment:
@AHeinisch hat es einen speziellen grund, warum die Animation mit der geringen Priorität ausgewählt wird und nicht die erste existierende Animation?

Als wir die Animationen noch direkt gesetzt haben, wurde über die übergebenen Animationen itteriert und die erste gesetzt, welche auch im Component gespeichert ist.

Reply:
@AHeinisch guck nochmal wie das genau ist.

Mein erwartetes verhalten wäre: Ich übergebe eine Menge an Animationen und die erste Animation die das DrawComponent auch hat, wird dann in die queue gehauen.